.github/workflows/go-lint.yml

```diff
@@ -6,35 +6,44 @@ on:
       go-version:
         required: false
         type: string
-        default: 'stable' # rc versions are defined like 1.24.0-rc.3
+        default: 'stable'
```

Comment:
Could default to what's in the go.mod instead.

Perhaps this can be made into a reusable workflow:

```yml
      - name: Extract Go version from go.mod
        id: go-version
        shell: bash
        run: |
          cd ${{ inputs.working-directory }}
          GO_VERSION=$(grep -E '^go [0-9]+\.[0-9]+' go.mod | awk '{print $2}')
          if [ -z "$GO_VERSION" ]; then
            echo "No Go version found in go.mod, using stable"
            GO_VERSION="stable"
          else
            echo "Go version $GO_VERSION found in go.mod"
          fi
          echo "version=$GO_VERSION" >> $GITHUB_OUTPUT
```